test.js

```diff
@@ -246,6 +246,61 @@ tape('persistent', function (t) {
   })
 })
 
+tape('getNodes', function (t) {
+  bootstrap(function (port, node) {
+    const a = dht({ bootstrap: port })
+    const b = dht({ bootstrap: port })
+    a.ready(function () {
+      b.ready(function () {
+        const aNodes = a.getNodes()
+        const bNodes = b.getNodes()
+        t.deepEqual([{ id: b.id, host: '127.0.0.1', port: b.address().port }], aNodes)
+        t.deepEqual([{ id: a.id, host: '127.0.0.1', port: a.address().port }], bNodes)
+        b.destroy()
+        a.destroy()
+        node.destroy()
+        t.end()
+      })
+    })
+  })
+})
+
+tape('addNodes', function (t) {
+  bootstrap(function (port, node) {
+    const a = dht({ bootstrap: port })
+    const b = dht({ bootstrap: [] })
+    b.listen() // https://github.com/hyperswarm/dht/issues/22
+
+    a.command('hello', {
+      query (data, callback) {
+        t.same(data.value, null, 'expected data')
+        callback(null, Buffer.from('world'))
+      }
+    })
+
+    a.ready(function () {
+      b.ready(function () {
+        setTimeout(function () {
```

Comment:
does process.nextTick work here or does it have to be a setTimeout?

Reply:
I considered trying that but did not because the docs said `nextTick` gets priority over the rest of the queue.

Reply:
Can you try? I think this is because the ready handler is called before the node is added.

Reply:
It works! What should I do?

Reply:
Nice! Replace setTimeout with that then. That much easier to reason about test wise :)

Reply:
Done!